### ISSUE TITLE
Question: If I change a table name, does the refs relevant to this table also change? And what about indexes.

### ISSUE THREAD

**Issue by idan3** [Vanderhoof/PyDBML#6]
As I checked. For Indexes, if I change a column name the index is not changed respectively. Do I do something wrong? Is it going to be fixed? Can someone tell me on what I can trust that will change respectively and on what not? (Same for enums)

**Comment by Vanderhoof** [comment]
Hi @idan3!
Refs and indexes have link to the parent Table object. If you change the name of the table, refs and indexes still point to this table object, even with the new name, so generated SQL and DBML for them should contain the new name.

 I did manual testing and it indeed works like this. Can you send me code examples where it doesn't?
 
 ```python
>>> parsed = PyDBML(Path('test_schema.dbml'))
>>> table = parsed.tables[2]
>>> table.name
'products'
>>> table.name = 'changed_products'  # changing table name
>>> table.indexes[0].sql  # table name changed in index
'CREATE INDEX "product_status" ON "changed_products" ("merchant_id", "status");'
>>> parsed.refs[1].sql  # table name changed in ref
'ALTER TABLE "order_items" ADD FOREIGN KEY ("product_id") REFERENCES "changed_products" ("id");'
```

**Comment by idan3** [comment]
Yes. But for example:
If I change an enum name the name won't be changed in the type of the column.
In the "Enum" class you should return "self" on the get_type function and not EnumType(self.name, self.items). And then, when you'll change enum types names, for example, it will be changed for all the columns using this enum. (see a pull request). I don't know why you use the simple str, I hope I don't miss something.
Another thing is Indexes. If I change a column name, the relevant refs are updated perfectly but indexes are not because you use 'subject_names' attribute (Which holds str type) when creating the dbml string from the parsed dbml instance instead of 'subject' which have a reference to the columns.

**Comment by Vanderhoof** [comment]
I think you are right about EnumType, we don't need this dummy class and should return Enum.
Regarding `subject_names`, we need it on parsing stage, because when index is parsed, we don't know anything about the columns, so can only operate on strings. This field shouldn't be used after parsing is done, but I see that it is used in `Index.dbml()` method (and in str/repr methods, which is not that critical). Interesting that in `Index.sql()` method I used `subjects`, as it should be done.

Let me do more testing on these issues and I'll fix them in 4.0.2

**Comment by idan3** [comment]
Ok. Try to read what I wrote about "subject". If I am not clear I can clarify.

By the way, great library!

**Comment by Vanderhoof** [comment]
I got it, these issues are pretty simple to fix. But I started digging and found some more problems with object editing that are more complicated.

For example, changing references. There's currently no proper way to change reference so that it points to another table or column. Even changing the reference name is inconsistent because we have two different objects for references: `Reference` which is stored in `ParsedResults.refs` and `TableReference` which is stored in `Table.refs`.

I think of a solution and publish a fix later.

Thanks for bringing this up : }

**Comment by idan3** [comment]
Good! I cant help if you need. 
In my opinion, there should be update end remove functions for almost every class (ex, table.remove(), or column.update(name="new_name")). That whole solve a lot and will be more clear to use. 'Update' is note mandatory but there is no way of removing.

**Comment by Vanderhoof** [comment]
I've published 0.4.2 with quick changes for enum and indexes. I didn't accept your PR because decided to do it differently — I completely removed the EnumType and this required to add a little change to PyDBMLParsedResults class.

As for editing references, I will rewrite the insides of PyDBML where this won't be an issue. The design was wrong from the start, when I started building this thing I just needed a parser and editing function was not even planned. 
I'll try to keep the interface the same so that it won't break other projects, but the concept has to be changed.
